src/users/controllers/CreateUserController.js

```diff
@@ -1,32 +1,34 @@
-import { Prisma } from '@prisma/client';
-import { prismaClient } from '../../../database/prismaClient.js';
-import { encryptPassword } from  '../utils/encrypt.js';
+import { Prisma } from "@prisma/client";
+import { prismaClient } from "../../../database/prismaClient.js";
+import { encryptPassword } from "../utils/encrypt.js";
+import { StatusCodes } from "http-status-codes";
 
 export class CreateUserController {
-    async handle(request, response) {
-        const {name, email, password} = request.body;
-        const hashPassword =  await encryptPassword(password)
-        try {
-            const user = await prismaClient.user.create({
-                data: {
-                    name,
-                    email,
-                    password: hashPassword
-                }
-            })
+  async handle(request, response) {
+    const { name, email, password } = request.body;
+    const hashPassword = await encryptPassword(password);
+    try {
+      const user = await prismaClient.user.create({
+        data: {
+          name,
+          email,
+          password: hashPassword,
+        },
+      });
 
-            return response.json({message:`Usuario criado com sucesso - ID: ${user.id}`})
-          
-        } catch (e) {
-            console.log(e);
-            if(e instanceof Prisma.PrismaClientKnownRequestError) {
-                
-                return response.status(e.errorCode).json({
-                    errorCode: e.errorCode,
-                    message: e.message
-                })
-            }
-            return response.json({e})
-        }
+      return response.status(StatusCodes.CREATED).json({
+        id: user.id,
+        name: user.name,
+      });
+    } catch (e) {
+      console.log(e);
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        return response.status(e.errorCode).json({
+          errorCode: e.errorCode,
+          message: e.message,
+        });
+      }
+      return response.json({ e });
```

Comment:
Geralmente a gente não expõe informações de error brutas no retorno da api. Vc não tem como saber com certeza se tem ou não informações sensíveis (não manjo de prisma).

Mas, como não tá em prod e vc está basicamente mandando o error inteiro em ambos casos, provavelmente não precisa desse if ai.

```js
return res
  .status(500)
  .json({
    code: e.errorCode,
    message: e.message
  })
```

Detalhe que, para conflito (usuário já existente), é interessante vc tartar isso antes de tentar criar o usuário.

```js
if (userExists) {
   return res
     .status(409)
     .json({
      message: "já exists"
     })
}
```

Importante considerar que, 409 sugere à um possível agressor do sistema que existe um usuário com o email informado. Em um sistema onde segurança é mais critico, uma alternativa seria retornar sucesso e pedir para confirmar o email:
- se o usuário não existia, cria o usuário e envia um email para confirmar o endereço de mail;
- se o usuário já existia, envia um email informando uma tentativa de acesso (ou algo assim) e sugere recuperar a senha (se isso for possível).

Acredito que para o nível de segurança que seu projeto demanda, um 409 tá bom já

Reply:
Opa, sim sim! Esse método eu vou refatorar, ele foi a implementação inicial sem a  camada de repository e afins! 
Anotei aqui já as melhorias